app/uk/gov/hmrc/incometaxsoftwarechoicesfrontend/controllers/GlossaryController.scala

```diff
@@ -89,4 +91,20 @@ class GlossaryController @Inject()(mcc: MessagesControllerComponents,
 
 }
 
+object GlossaryController extends HeaderNames {
+  def referringProvider(implicit request: Request[AnyContent]): Option[String] = {
+    request.headers.get(REFERER).flatMap(r => {
+      val url = new URL(r)
+      val path = URLDecoder.decode(url.getPath, "UTF-8")
+      val softwareProvider = path.split("/").reverse.head
+      val url1 = uk.gov.hmrc.incometaxsoftwarechoicesfrontend.controllers.routes.ProductDetailsController.show(softwareProvider).url
+      val loopbackCheck = url1 == path
```

Comment:
This is the check that the referer header is _exactly_ that of some vendor detail page.  If you can suggest any improvements, feel free....